Introduce TermMeta & TermMetaAdminUI class

grappler opened the pull request.

grappler committed:
Correct typos for post meta and sanitize value by default

sanitize_text_field() seemed to be the most common sanitization function. It ca be easy to forget to override the sanitization method. It better to have the wrong sanitization instead of none.

grappler committed:
Introduce TermMeta class

grappler committed:
Introduce TermMetaAdminUI class

ocean90 commented on src/PostMeta.php:
```diff
@@ -123,7 +123,7 @@ public function auth( $allowed, $meta_key, $object_id, $user_id, $cap, $caps ):
 	 */
 	public function sanitize( $meta_value, $meta_key, $object_type ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		return $meta_value;
+		return sanitize_text_field( $meta_value );
```
The objects should reflect WordPress's default configs, by default no sanitization is applied.

```suggestion
		return $meta_value;
```

ocean90 commented on src/TermMeta.php:
```diff
@@ -0,0 +1,150 @@
+	 * @return mixed Sanitized meta value.
+	 */
+	public function sanitize( $meta_value, $meta_key, $object_type ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		return sanitize_text_field( $meta_value );
```
The objects should reflect WordPress's default configs, by default no sanitization is applied.

```suggestion
		return $meta_value;
```

ocean90 commented on src/TermMetaAdminUI.php:
```diff
@@ -0,0 +1,143 @@
+use WP_Term;
+
+/**
+ * Class used to implement term meta.
```
```suggestion
 * Class used to implement a UI for adding/editing term meta.
```

ocean90 commented on src/TermMeta.php:
```diff
@@ -0,0 +1,150 @@
+ *
+ * @since 1.0.0
+ *
+ * @package Required\Common
```
```suggestion
```

See https://github.com/wearerequired/coding-standards/issues/28.

ocean90 commented on src/TermMetaAdminUI.php:
```diff
@@ -0,0 +1,143 @@
+ *
+ * @since 1.0.0
+ *
+ * @package Required\Common
```
```suggestion
```

ocean90 commented on src/TermMetaAdminUI.php:
```diff
@@ -0,0 +1,143 @@
+	 * Renders field when adding a new term.
+	 *
+	 * @since 1.0.0
+	 *
```
```suggestion
```

ocean90 committed:
Inline docs improvements

Co-Authored-By: grappler <grapplerulrich@gmail.com>

ocean90 reviewed (changes requested):
Changing sanitize() is still missing.

ocean90 committed:
Objects should reflect WordPress's default configs

The objects should reflect WordPress's default configs, by default no sanitization is applied.

Co-Authored-By: grappler <grapplerulrich@gmail.com>

grappler commented:
This is how I would register the term meta admin ui.
```php
add_action( 'admin_init', [ $this, 'register_term_meta_admin_ui' ] );
/**
 * Registers term meta admin ui.
 *
 * @since 1.0.0
 */
public function register_term_meta_admin_ui() {
	$term_meta_admin_ui = new Container();

	$term_meta_admin_ui->add(
		TermMeta\SchoolDistrictPostalCode::KEY,
		new TermMeta\SchoolDistrictPostalCodeAdminUI(
			Taxonomy\SchoolDistrict::NAME,
			TermMeta\SchoolDistrictPostalCode::KEY
		)
	);

	$term_meta_admin_ui->bootstrap();
}
```

ocean90 commented on src/TermMetaAdminUI.php:
```diff
@@ -37,10 +37,10 @@ abstract class TermMetaAdminUI implements Registrable {
+	 * @param string $term_meta The term meta key.
 	 */
-	public function __construct( Taxonomy $taxonomy, TermMeta $term_meta ) {
+	public function __construct( string $taxonomy, string $term_meta ) {
```
I'm not supporting this change. Dependency injection is preferred as it give's us benefits like type hinting.

grappler replied:
I can understand that. I would prefer to separate the registration of `TermMetaAdminUI` from the Taxonomy registration. What I see is also that additional information in `TermMeta` is registered differently with `taxonomy()` instead of using the construct. I would be good to do it the same everywhere.
